crates/rooch-types/src/transaction/ethereum.rs

```diff
@@ -26,11 +26,51 @@ use serde::{Deserialize, Serialize};
 pub struct EthereumTransaction(pub ethers::core::types::Transaction);
 
 impl EthereumTransaction {
-    //This function is just a demo, we should define the Ethereum calldata's MoveAction standard
+    // This function is just a demo, we should define the Ethereum calldata's MoveAction standard
     pub fn decode_calldata_to_action(&self) -> Result<MoveAction> {
-        //Maybe we should use RLP to encode the MoveAction
-        bcs::from_bytes(&self.0.input)
-            .map_err(|e| anyhow::anyhow!("decode calldata to action failed: {}", e))
+        if self.0.input.is_empty() {
+            //let account_address = self.rpc_service.resolve_address(self.sender()).await?;
+
+            let test = TxContext::random_for_testing_only();
+            let serialized = test.to_bytes();
+
+            // Handle the transfer gas coin function call only in ethereum
+            let payload = MoveAction::new_function_call(
+                moveos_types::move_types::FunctionId::new(
+                    move_core_types::language_storage::ModuleId::new(
+                        AccountAddress::from_hex_literal("0x3").unwrap(),
+                        Identifier::new("coin").unwrap(),
+                    ),
+                    Identifier::new("transfer").unwrap(),
```

Comment:
The GasCoin currently does not support public transfer. If we want to support transfer coin, we need to add the `store` ability to the `GasCoin` struct. Or provide a `gas_coin::transfer` function.